src/dataworker/Dataworker.ts

```diff
@@ -13,40 +21,134 @@ export class Dataworker {
+      if (!originClient.isUpdated())
+        throw new Error(`origin spokepoolclient with chain ID ${originChainId} not updated`);
+
+      // Loop over all other SpokePoolClient's to find deposits whose destination chain is the selected origin chain.
+      this.logger.debug({ at: "Dataworker", message: `Looking up data for origin spoke pool`, originChainId });
+      for (const destinationChainId of Object.keys(this.spokePoolClients)) {
+        if (originChainId === destinationChainId) continue;
+
+        const destinationClient = this.spokePoolClients[destinationChainId];
+        if (!destinationClient.isUpdated())
+          throw new Error(`destination spokepoolclient with chain ID ${destinationChainId} not updated`);
+
+        // Store deposits whose destination chain is the selected chain as unfilled deposits and set the initial fill
+        // amount remaining equal to the full deposit amount minus any valid fill amounts.
+        // Remove any deposits that have no unfilled amount (i.e that have an unfilled amount of 0) and append the
+        // remaining deposits to the unfilledDeposits array.
+        const depositsForDestinationChain = originClient.getDepositsForDestinationChain(destinationChainId);
+        this.logger.debug({
+          at: "Dataworker",
+          message: `Found ${depositsForDestinationChain.length} deposits for destination chain ${destinationChainId}`,
+          originChainId,
+          destinationChainId,
+        });
+
+        const unfilledDepositsForDestinationChain = depositsForDestinationChain
+          .map((deposit) => {
+            return { data: deposit, unfilledAmount: destinationClient.getValidUnfilledAmountForDeposit(deposit) };
+          })
+          .filter((deposit) => deposit.unfilledAmount.gt(0));
+
+        if (unfilledDepositsForDestinationChain.length > 0) {
+          assign(unfilledDeposits, [destinationChainId], unfilledDepositsForDestinationChain);
+        } else {
+          this.logger.debug({
+            at: "Dataworker",
+            message: `All deposits are filled`,
+            originChainId,
+            destinationChainId,
+          });
+        }
+
+        // Grab all valid fills submitted to the destination spoke pool.
+        const fillsOnDestinationChain = destinationClient.getFills();
+        const validFillsOnDestinationChain = fillsOnDestinationChain.filter((fill) => {
+          // For each fill, see if we can find a deposit sent from the origin client that matches it.
+          for (const deposit of depositsForDestinationChain) {
+            // @dev: It doesn't matter which client we call validateFillForDeposit() on as the logic is
+            // chain agnostic.
+            // @dev: All of the deposits returned by `getDepositsForDestinationChain` will include the expected realized
+            // lp fee % for the deposit quote time. If this fill does not have the same realized lp fee %, then it will
+            // be ignored.
```

Comment:
this commenting is non-standard with the `@dev` flag. we should not mix and match commenting styles methinks.
```suggestion
            // Note: it doesn't matter which client we call validateFillForDeposit() on as the logic is chain agnostic.
            // All of the deposits returned by getDepositsForDestinationChain will include the expected realized
            // lp fee % for the deposit quote time. If this fill does not have the same realized lp fee %, then it will be ignored.
```